src/js/utils/date-utils.js

```diff
@@ -0,0 +1,289 @@
+         *
+        */
+        DateObject = function(datetime) {
+            if (datetime === undefined || datetime === '') {
+                return '';
+            }
+            return MomentLib(datetime).utc();
+        };
+
+        /**
+         * attempt to determine user's timezone
+         * from preferences, then browser settings
+         *
+         * @param {string} timezone representation, 'Region/Population_Center' e.g. 'America/New_York'.
+         * If this is None or undefined, we attempt to determine the TZ from system/browser settings
+         *
+         * @return {string} either the entered or determined
+         * timezone, or (in the case of default) UTC
+         *
+         */
+        UserTimeZone = function(timezone) {
+            // linter wants a different variable
+            var returnzone;
+            if (timezone === undefined || timezone === 'None') {
+                returnzone = MomentTZ.tz.guess();
+            } else {
+                returnzone = timezone;
+            }
+            if (returnzone === undefined) {
+                return 'UTC';
+            }
+            return returnzone;
+        };
+
+        /**
+         *
+         * Attempt to transform time based on user pref/sys timezone
+         *
+         * @param {Object} dateobject, a moment.js datetime object
+         * @param {string} usertz, a timezone representation, e.g. 'America/New_York'
+         *
+         * @return {object} a moment.js datetime object in a determined timezone
+         */
+        LocalizedTime = function(dateobject, usertz) {
+            if (dateobject === undefined) {
+                return '';
+            }
+            if (usertz === undefined) {
+                return dateobject;
+            }
+            return dateobject.tz(usertz);
+        };
+
+        /**
+         * This generates the TZ-aware and language-pref displayed time string,
+         * with a set of built-ins for standard format across the platform
+         *
+         * the formatting is extensible, but this should suffice for most use cases
+         *
+         * e.g.
+         * preferred_language = 'en' yields Oct. 14, 2016 09:00 BST
+         * preferred_language = 'ru' yields 14 окт 2016 г. 09:00 BST
+         *
+         * Default to US-English
+         * (which is the current unlocalized view)
+         *
+         * @param {object} localizedtime a moment.js datetime object in a determined timezone
+         * @param {string} language: an ISO 639-1 language string (e.g. 'en', 'fr')
+         * @param {string} format: (optional) a desired format constant
+         *
+         * @return {string} a language/timezone aware localized string
+        */
+        TimeDisplay = function(localizedtime, language, format) {
+            /**
+             * This is a function to determine the language based on browser settings,
+             * but should probably be overridden by platform-set language preference
```

Comment:
That's awkwardly worded. I meant that user pref-lang override the browser settings (like in the timezones). Fixed.